test/parallel/test-cluster-worker-wait-server-close.js

```diff
@@ -0,0 +1,63 @@
+var common = require('../common');
+var assert = require('assert');
+var cluster = require('cluster');
+var net = require('net');
+
+if (cluster.isWorker) {
+  net.createServer(function(socket) {
+    // Wait for any data, then close connection
+    socket.on('data', socket.end.bind(socket));
+  }).listen(common.PORT, common.localhostIPv4);
+} else if (cluster.isMaster) {
+
+  var connectionDone;
+  var checks = {
+    disconnectedOnClientsEnd: false,
+    workerDied: false
+  };
+
+  // helper function to check if a process is alive
+  var alive = function(pid) {
+    try {
+      process.kill(pid, 0);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  };
+
+  // start worker
+  var worker = cluster.fork();
+
+  // Disconnect worker when it is ready
+  worker.once('listening', function() {
+    net.createConnection(common.PORT, common.localhostIPv4, function() {
+      var socket = this;
+      setTimeout(function() {
+        worker.disconnect();
+        setTimeout(function() {
+          socket.write('.');
+          connectionDone = true;
+        }, 1000);
+      }, 1000);
+    });
+  });
+
+  // Check worker events and properties
+  worker.once('disconnect', function() {
+    checks.disconnectedOnClientsEnd = connectionDone;
+  });
+
+  // Check that the worker died
+  worker.once('exit', function() {
```

Comment:
This test is a no-op. It confirms what the `child_process` exit event does not occur until the child process has exited. It could be that breaks some day, but it will be a bug in `child_process`, unrelated to cluster, and wholly unrelated to what this test is looking at.